shared/js/media/media_frame.js

```diff
@@ -245,7 +247,20 @@ MediaFrame.prototype.displayImage = function displayImage(blob,
     }
   }
   else {
-    noPreview();
+    // If we have valid small EXIF preview use it for css background image
+    // This will help to load large downsampled preview images without flash
+    // See bug 1188286
+    // ( TBD: preview & preview.start might not be enough.
+    // Need more robust check for small valid EXIF preview )
+    if (preview && preview.start) {
```

Comment:
I think maybe I'd combine this if with the else above to have an else if and I'd do
else if (preview && preview.width && preview.height && preview.start) {
Note that if you don't check the aspect ratio then it might be possible to get a flash of a distorted preview before the full image loads, and that would probably be bad.  I actually have some test images like this in apps/gallery/test/images/: images where the EXIF preview has the wrong aspect ratio and so is not used. This patch would make them display that bad preview briefly.

So maybe refactor usePreview() into two methods. The first would just be isValidPreview() that would do all the checks  (preview exists, size is non-zero, aspect ratio is correct) except for the size is big enough check.  The second would be isBigEnoughPreview() which would do the size check. Then the code here could be refactored to look like this:
```
var isValid = isValidPreview(preview);
var isBigEnough = isBigEnoughPreview(preview);
if (isValid && isBigEnough) {
  // this is the existing usePreview() code
}
else if (isValid) {
  // this is your new case where you're setting the background image.
}
else {
  // this is the existing nopreview case
}
```
if (isValidPreview(preview) && isBigEnough(preview))